nfs/src/rpc_task.cpp

```diff
@@ -2761,6 +2792,18 @@ static void readdir_callback(
         dircache_handle->set_cookieverf(&res->READDIR3res_u.resok.cookieverf);
 
         if (eof) {
+            if (!has_new_entry) {
+                /*
+                 * We have reached eof however, we have not seen any new
+                 * entries, this means we are reenumerating and the directory
+                 * has shrunk after we started. Since, we have to respond to the
+                 * original fuse call, we need to send an error here.
+                 */
+                
+                task->reply_error(EIO);
```

Comment:
why do we need to fail? Why not return eof, as that's what has truly happened.
I'd suggest try to emulate this and see how Linux NFS client behaves and emulate that beahviour.